dirty_equals/_other.py

```diff
@@ -145,3 +146,64 @@ def is_even(x):
 
     def equals(self, other: Any) -> bool:
         return self.func(other)
+
+
+IP = TypeVar('IP', IPv4Address, IPv4Network, IPv6Address, IPv6Network, Union[str, int, bytes])
+
+
+class IsIP(DirtyEquals[IP]):
+    """
+    A class that checks if a value is a valid IP address, optionally checking IP version, netmask.
+    """
+
+    def __init__(self, *, version: Literal[None, 4, 6] = None, netmask: Optional[str] = None):
+        """
+        Args:
+            version: The version of the IP to check, if omitted, versions 4 and 6 are both accepted.
+            netmask: The netmask of the IP to check, if omitted, any netmask is accepted. Requires version.
+
+        ```py title="IsIP"
+        from ipaddress import IPv4Address, IPv6Address, IPv4Network
+        from dirty_equals import IsIP
+
+        assert '179.27.154.96' == IsIP
+        assert '179.27.154.96' == IsIP(version=4)
+        assert '2001:0db8:0a0b:12f0:0000:0000:0000:0001' == IsIP(version=6)
+        assert IPv4Address('127.0.0.1') == IsIP
+        assert IPv4Network('43.48.0.0/12') == IsIP
+        assert IPv6Address('::eeff:ae3f:d473') == IsIP
+        assert '54.43.53.219/10' == IsIP(version=4, netmask='255.192.0.0')
+        assert '54.43.53.219/10' == IsIP(version=4, netmask=4290772992)
+        assert '::ffff:aebf:d473/12' == IsIP(version=6, netmask='fff0::')
+        assert 3232235521 == IsIP
+        ```
+        """
+        self.version = version
+        if netmask and not self.version:
+            raise TypeError('To check the netmask you must specify the IP version')
+        self.netmask = netmask
+        super().__init__(version=version or Omit, netmask=netmask or Omit)
+
+    def equals(self, other: Any) -> bool:
+
+        if isinstance(other, (IPv4Network, IPv6Network)):
+            ip = other
+        elif isinstance(other, (str, bytes, int, IPv4Address, IPv6Address)):
+            ip = ip_network(other, strict=False)
+        else:
+            return False
+
+        return self.run_kwargs_checks(ip)
+
+    def run_kwargs_checks(self, ip: Union[IPv4Network, IPv6Network]) -> bool:
```

Comment:
Any reason this should be a separate method?

Reply:
Not really, think I left this over from a previous draft - will tweak